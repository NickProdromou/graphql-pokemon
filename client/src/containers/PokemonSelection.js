import React, { Component } from "react";
import Styled from 'styled-components';
import { Page, Column, Row } from "hedron";

import { spacing, colours } from "../style/variables";
import { type } from "../style/mixins/index";

class PokemonSelection extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstPokemonSelected: true,
      secondPokemonSelected: true
    };
  }

  render() {
    const { firstPokemonSelected, secondPokemonSelected } = this.state;

    return (
      <SelectionSection>
        <Page>
          <Row className="SelectionRow">
            <Column lg={6}>
              <div className="SelectionInner">
                <h2 className="selectionTitle">pokemon 1</h2>
                <p>AutocompleteSearchBar</p>
                <p>PokemonCard</p>
              </div>              
            </Column>
            <Column lg={6}>
              <div className="SelectionInner">
                <h2 className="selectionTitle">pokemon 2</h2>
                <p>AutocompleteSearchBar</p>
                <p>PokemonCard</p>
              </div>              
            </Column>
          </Row>
          <Row>
            <Column>
              <div className="SelectionCompareButton">
                {firstPokemonSelected &&
                secondPokemonSelected && <button>Compare</button>}
              </div>
            </Column>
          </Row>
        </Page>
      </SelectionSection>
    );
  }
}

const SelectionSection = Styled.div`  
  flex: 1;

  .SelectionInner, .SelectionCompareButton {
    padding: 0 ${spacing.mid.level1};    
  }

  .selectionTitle {
    ${type('heading2')}
    color: ${colours.accentColor}
  }

`;

export default PokemonSelection;
