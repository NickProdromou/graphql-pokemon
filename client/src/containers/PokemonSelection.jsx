import React, { Component } from "react";
import Styled from "styled-components";
import { Page, Column, Row } from "hedron";

import { spacing, colours } from "../style/variables";
import { type } from "../style/mixins/index";

import PokemonSearch from "../components/PokemonSearch";
import PokemonCard from "../components/PokemonCard";

class PokemonSelection extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstPokemonSelected: "",
      secondPokemonSelected: ""
    };
  }

  checkChildInput = (value, identifier) => {
    this.setState({
      [identifier]: value.suggestion
    });
  };

  render() {
    const { firstPokemonSelected, secondPokemonSelected } = this.state;

    return (
      <SelectionSection>
        <Page>
          <Row className="SelectionRow">
            <Column lg={6}>
              <div className="SelectionInner">
                <PokemonSearch
                  inform={this.checkChildInput}
                  identifier="firstPokemonSelected"
                />
                {firstPokemonSelected && (
                  <PokemonCard pokemonName={this.state.firstPokemonSelected} />
                )}
              </div>
            </Column>
            <Column lg={6}>
              <div className="SelectionInner">
                <PokemonSearch
                  inform={this.checkChildInput}
                  identifier="secondPokemonSelected"
                />
                {secondPokemonSelected && (
                  <PokemonCard pokemonName={this.state.secondPokemonSelected} />
                )}
              </div>
            </Column>
          </Row>
          <Row>
            <Column>
              <div className="SelectionCompareButtonRow">
                {firstPokemonSelected &&
                  secondPokemonSelected &&
                  firstPokemonSelected !== secondPokemonSelected && (
                    <button className="SelectionCompareButton">See Comparison Chart</button>
                  )}
              </div>
            </Column>
          </Row>
        </Page>
      </SelectionSection>
    );
  }
}

const SelectionSection = Styled.div`  
  flex: 1;

  .SelectionInner, .SelectionCompareButtonRow {
    padding: 0 ${spacing.mid.level1};    
  }

  .selectionTitle {
    ${type("heading2")}
    color: ${colours.accentColor};
  }

  .SelectionCompareButton {
    ${type('ui')}
    background: ${colours.accentColor};    
    color: ${colours.primaryColorLight};
    padding: ${spacing.small.level2} ${spacing.small.level3};
  }

`;

export default PokemonSelection;
