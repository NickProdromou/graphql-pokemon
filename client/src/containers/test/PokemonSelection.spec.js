import React from "react";
import renderer from "react-test-renderer";
import PokemonSelection from "../PokemonSelection";

describe("PokemonSelection Component", () => {
  test("It renders without error", () => {
    const component = renderer.create(<PokemonSelection />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
