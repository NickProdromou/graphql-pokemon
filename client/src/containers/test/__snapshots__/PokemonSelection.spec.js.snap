// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonSelection Component It renders without error 1`] = `
<div
  className="sc-ifAKCX dmHmU"
>
  <div
    className="sc-bxivhb bIAnAi"
  >
    <section
      className="SelectionRow sc-bwzfXH irrVft"
    >
      <div
        className="sc-htpNat hrIxvT"
      >
        <div
          className="SelectionInner"
        >
          <h2
            className="selectionTitle"
          >
            pokemon 1
          </h2>
          <p>
            AutocompleteSearchBar
          </p>
          <p>
            PokemonCard
          </p>
        </div>
      </div>
      <div
        className="sc-htpNat hrIxvT"
      >
        <div
          className="SelectionInner"
        >
          <h2
            className="selectionTitle"
          >
            pokemon 2
          </h2>
          <p>
            AutocompleteSearchBar
          </p>
          <p>
            PokemonCard
          </p>
        </div>
      </div>
    </section>
    <section
      className="sc-bwzfXH irrVft"
    >
      <div
        className="sc-htpNat bKOCBb"
      >
        <div
          className="SelectionCompareButton"
        >
          <button>
            Compare
          </button>
        </div>
      </div>
    </section>
  </div>
</div>
`;
